### PR TITLE
break server into middleware, add metadata rendering

### PR DESCRIPTION
**Pull request by Adjective-Object**
address #35

Had to add a case to the irc log, because the normalized irc log format uses messages with empty lines sometimes.

The server module was getting large, so I broke it down into metadata & state prepop modules.

### CODE REVIEW

**Comment by nolanlum** [comment]
this broke permalinks https://qdb.esports.moe/quote/13

i'm not sure why.

File: src/server/metadata.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+                });
+
+            next();
+        };
+
+        // fetch the quote if it was not in the state, otherwise
+        // use the stored quote in the next stage of middleware.
+        if (!quote) {
+            api_handle.qdbQuoteGetById({quoteId: id})
+                .then()
```
Comment (nolanlum) [inline src/server/metadata.tsx]: this doesn't seem good

Comment (Adjective-Object) [inline src/server/metadata.tsx]: it's unnecessary but shouldn't actually effect anything, catch catches things from upstream on the whole chain

Comment (Adjective-Object) [inline src/server/metadata.tsx]: Can we get an example request to test locally with?

**Pull request merged**